sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScan.scala

```diff
@@ -71,7 +103,7 @@ abstract class FileScan(
   }
 
   protected def partitions: Seq[FilePartition] = {
-    val selectedPartitions = fileIndex.listFiles(Seq.empty, Seq.empty)
+    val selectedPartitions = fileIndex.listFiles(partitionFilters, Seq.empty)
```

Comment:
@gengliangwang @cloud-fan continuing the discussion from above (the comment was on the wrong line).
The `V2ScanRelationPushDown` rule will pushdown the `dataFilters` only to datasources which support pushdown by implementing the `SupportsPushDownFilters`.
Datasources such as `csv` and `json` do not implement the `SupportsPushDownFilters` trait. In order to support data skipping uniformly for all file based data sources, we override the `listFiles` method in a FileIndex implementation, which consults external metadata and prunes the list of files.
The suggestion is to make the necessary changes to have the `dataFilters` passed to the `listFiles` as well.
Otherwise, one would have to create a new datasource implementation in order to support each file based datasource that doesn't have a built in pushdown mechanism.

Reply:
This makes sense to me. @gengliangwang what do you think?

At least you can disable v2 file source to bring back this feature.

Reply:
Yes, it makes sense if there is a `fileIndex` can use the `dataFilters`.
@guykhazma  could you create a PR for this?

Reply:
@gengliangwang @cloud-fan sure, thanks.
I have opened this [PR](https://github.com/apache/spark/pull/27157)